src/currentprocess.rs

```diff
@@ -140,37 +142,8 @@ impl home::env::Env for Process {
     }
 }
 
-#[cfg(feature = "test")]
-impl From<TestProcess> for Process {
-    fn from(p: TestProcess) -> Self {
-        Self::TestProcess(p)
-    }
-}
-
 static HOOK_INSTALLED: Once = Once::new();
 
-/// Run a function in the context of a process definition.
-///
-/// If the function panics, the process definition *in that thread* is cleared
-/// by an implicitly installed global panic hook.
-pub fn with<F, R>(process: Process, f: F) -> R
-where
-    F: FnOnce() -> R,
-{
-    ensure_hook();
-
-    PROCESS.with(|p| {
-        if let Some(old_p) = &*p.borrow() {
-            panic!("current process already set {old_p:?}");
-        }
-        let _guard = crate::cli::log::tracing_subscriber(&process).set_default();
-        *p.borrow_mut() = Some(process);
-        let result = f();
-        *p.borrow_mut() = None;
-        result
-    })
-}
-
 fn ensure_hook() {
```

Comment:
A question: maybe I've missed it, but is `ensure_hook()` called anywhere? Somehow my search box didn't catch it 🤔

Reply:
This PR doesn't get rid of `with_runtime()` yet, which uses it. Next on the list!

Reply:
> This PR doesn't get rid of `with_runtime()` yet, which uses it. Next on the list!

@djc Do you think it's still valuable to call it though? To be frank I'm not sure what this function is doing, so...